### PR TITLE
added support for automl-models

### PR DESCRIPTION
**Pull request by ifielker**
added support for automl-models

### CODE REVIEW

File: firebase_admin/ml.py [hunk 1]
```diff
@@ -52,6 +52,9 @@
 _TAG_PATTERN = re.compile(r'^[A-Za-z0-9_-]{1,60}$')
 _GCS_TFLITE_URI_PATTERN = re.compile(
     r'^gs://(?P<bucket_name>[a-z0-9_.-]{3,63})/(?P<blob_name>.+)$')
+_AUTO_ML_MODEL_PATTERN = re.compile(
+    r'^projects/(?P<project_id>\d+)/locations/(?P<location_id>[^/]+)/' +
```
Comment (hiranya911) [inline firebase_admin/ml.py]: Project ID group is defined as `(?P<project_id>[a-z0-9-]{6,30})` in some of the other regex patterns in this file.

File: firebase_admin/ml.py [hunk 2]
```diff
@@ -363,9 +366,15 @@ def from_dict(cls, data):
         """Create an instance of the object from a dict."""
         data_copy = dict(data)
         model_source = None
+
         gcs_tflite_uri = data_copy.pop('gcsTfliteUri', None)
         if gcs_tflite_uri:
             model_source = TFLiteGCSModelSource(gcs_tflite_uri=gcs_tflite_uri)
+
+        auto_ml_model = data_copy.pop('automlModel', None)
```
Comment (hiranya911) [inline firebase_admin/ml.py]: Define a helper function for all this.

```
def _init_model_source(data_copy):
  gcs_tflite_uri = ...
  if gcs_tflite_uri:
    return ...

  auto_ml_model = ....
  if gcs_tflite_uri:
    return ...
  return None
```

File: firebase_admin/ml.py [hunk 3]
```diff
@@ -592,6 +601,36 @@ def as_dict(self, for_upload=False):
...
+class TFLiteAutoMlSource(TFLiteModelSource):
+    """TFLite model source representing a tflite model created via AutoML."""
+
+    def __init__(self, auto_ml_model, app=None):
+        self._app = app
+        self._auto_ml_model = _validate_auto_ml_model(auto_ml_model)
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self._auto_ml_model == other._auto_ml_model  # pylint: disable=protected-access
```
Comment (hiranya911) [inline firebase_admin/ml.py]: Just access the public property and get rid of the linter directive.

File: firebase_admin/ml.py (hunk 1 above)
Comment (ifielker) [reply firebase_admin/ml.py]: Done.

File: firebase_admin/ml.py (hunk 2 above)
Comment (ifielker) [reply firebase_admin/ml.py]: Done

File: firebase_admin/ml.py (hunk 3 above)
Comment (ifielker) [reply firebase_admin/ml.py]: Done

File: firebase_admin/ml.py [hunk 4]
```diff
@@ -592,6 +601,36 @@ def as_dict(self, for_upload=False):
         return {'gcsTfliteUri': self._gcs_tflite_uri}
 
 
+class TFLiteAutoMlSource(TFLiteModelSource):
+    """TFLite model source representing a tflite model created via AutoML."""
+
+    def __init__(self, auto_ml_model, app=None):
+        self._app = app
+        self._auto_ml_model = _validate_auto_ml_model(auto_ml_model)
```
[the remark this answers is not in the document]
Comment (ifielker) [reply firebase_admin/ml.py]: Done

File: firebase_admin/ml.py [hunk 5]
```diff
@@ -389,6 +378,17 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
+    @staticmethod
+    def _init_model_source(data):
+        model_source = None
+        gcs_tflite_uri = data.pop('gcsTfliteUri', None)
+        if gcs_tflite_uri:
+            model_source = TFLiteGCSModelSource(gcs_tflite_uri=gcs_tflite_uri)
```
Comment (hiranya911) [inline firebase_admin/ml.py]: return

File: firebase_admin/ml.py [hunk 6]
```diff
@@ -389,6 +378,17 @@ def __eq__(self, other):
...
+        auto_ml_model = data.pop('automlModel', None)
+        if auto_ml_model:
+            model_source = TFLiteAutoMlSource(auto_ml_model=auto_ml_model)
```
Comment (hiranya911) [inline firebase_admin/ml.py]: return

File: firebase_admin/ml.py [hunk 7]
```diff
@@ -606,11 +606,11 @@ class TFLiteAutoMlSource(TFLiteModelSource):
 
     def __init__(self, auto_ml_model, app=None):
         self._app = app
-        self._auto_ml_model = _validate_auto_ml_model(auto_ml_model)
+        self.auto_ml_model = auto_ml_model
 
     def __eq__(self, other):
         if isinstance(other, self.__class__):
-            return self._auto_ml_model == other._auto_ml_model  # pylint: disable=protected-access
+            return self._auto_ml_model == other.auto_ml_model
```
Comment (hiranya911) [inline firebase_admin/ml.py]:
```suggestion
            return self.auto_ml_model == other.auto_ml_model
```

File: firebase_admin/ml.py (hunk 5 above)
Comment (ifielker) [reply firebase_admin/ml.py]: done

File: firebase_admin/ml.py (hunk 6 above)
Comment (ifielker) [reply firebase_admin/ml.py]: done

File: firebase_admin/ml.py (hunk 7 above)
Comment (ifielker) [reply firebase_admin/ml.py]: done

**Pull request merged**